Modules/MessageUtils/PaginatedEmbed.js

```diff
@@ -1,75 +1,76 @@
 const { Colors, PageEmojis } = require("../../Internals/Constants");
 
 class PaginatedEmbed {
-	constructor (msg, descriptions = [], embed = null, images = [], fields = [], thumbnails = []) {
-		this.client = msg.client;
-		this.originalMsg = msg;
-
+	/**
+	 * After creating a PaginatedEmbed call `#init()` to set it up and start listenig for reactions.
+	 *
+	 * @param originalMsg 	The original message that created this paginated embed.
+	 * 						May be a custom object, the only required fields are `channel` and `author.id`
+	 * @param embedTemplate A slightly edited embed object that serves as the base template for all pages,
+	 * 						with strings being formatted via templates
+	 * @param pageData		All the data used for the different pages of the embed pages,
+	 * 						with the fields being arrays with values (or null) for every page
+	 */
+	constructor (originalMsg, embedTemplate, pageData) {
+		this.channel = originalMsg.channel;
+		this.authorID = originalMsg.author.id;
 		this.pageEmojis = PageEmojis;
 		this.pageEmojiArray = [...Object.values(this.pageEmojis)];
 
-		this.descriptions = descriptions;
-		this.currentDescription = 0;
-		this.totalDescriptions = this.descriptions.length - 1;
+		this.authors = pageData.authors || [];
+		this.titles = pageData.titles || [];
+		this.colors = pageData.colors || [];
+		this.urls = pageData.urls || [];
+		this.descriptions = pageData.descriptions || [];
+		this.fields = pageData.fields || [];
+		this.timestamps = pageData.timestamps || [];
+		this.thumbnails = pageData.thumbnails || [];
+		this.images = pageData.images || [];
+		this.footers = pageData.footers || [];
 
 		this.embedTemplate = Object.assign({
-			title: ``,
+			author: "{author}",
+			authorIcon: null,
+			authorUrl: null,
+			title: "{title}",
 			color: Colors.INFO,
-			description: `{description}`,
-			footer: `Embed {current description} out of {total descriptions} embeds`,
-		}, embed);
-		this.images = images;
-		this.fields = fields;
-		this.thumbnails = thumbnails;
+			url: null,
+			description: "{description}",
+			fields: null,
+			timestamp: null,
+			thumbnail: null,
+			image: null,
+			footer: "{footer}Page {currentPage} out of {totalPages}",
+			footerIcon: null,
+		}, embedTemplate);
+
+		this.currentPage = 0;
+		this.totalPages = pageData.pageCount || this.descriptions.length;
 	}
 
 	async init (timeout = 300000) {
-		await this.sendInitialMessage();
-		if (this.descriptions.length !== 1) {
+		await this._sendInitialMessage();
+		if (this.totalPages > 1) {
 			this.collector = this.msg.createReactionCollector(
-				(reaction, user) => user.id === this.originalMsg.author.id && this.pageEmojiArray.includes(reaction.emoji.name),
+				(reaction, user) => user.id === this.authorID && this.pageEmojiArray.includes(reaction.emoji.name),
```

Comment:
In a perfect world we'd also check the msg ids so they match (prevent fuckups.exe)

Reply:
Discord.JS does that internally already.